### PR TITLE
[14.0][ADD] mrp_attachment_mgmt: New addon.

### PR DESCRIPTION
**Pull request by victoralmau**
This module allows to get all attachments of a bill of materials (all levels) on a view.
This module allows to get all attachments of a work orders products to produce on a view.

Please @pedrobaeza and  @chienandalu can you review it?

@Tecnativa TT33396

### CODE REVIEW

File: mrp_attachment_mgmt/README.rst [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+Usage
+=====
+
+#. Go to *Manufacturing -> Operation > Work Orders* and click in window open button (after the "Start" and "Block" options).
+#. The smart-button "Attachments" display the attachments of the related product.
+
+
+#. Go to *Manufacturing -> Products > Bill of material -> Desk Combination*.
+#. Go to "Attachment" icon related to "Office Chair Black" component and upload some file.
```
Comment (chienandalu) [inline mrp_attachment_mgmt/README.rst:51]: I get this error:
```
int() argument must be a string, a bytes-like object or a number, not 'NoneType'
```
Peek 08-03-2022 12-40

**Review by chienandalu — CHANGES_REQUESTED** [review]

File: mrp_attachment_mgmt/README.rst (hunk 1 above)
Comment (victoralmau) [reply mrp_attachment_mgmt/README.rst:51]: Sorry, it is necessary to upload the files from the other icon and use the smart-button only to see all the attachments.
ejemplo

**Review by chienandalu — APPROVED** [review]
Perfect! :+1:

**Comment by pedrobaeza** [comment]
We can't it be done from both places? I think that a not good UX experience. Fix it for being done in both places, or remove the smart-button, or avoid the create button there.

**Comment by victoralmau** [comment]
> We can't it be done from both places? I think that a not good UX experience. Fix it for being done in both places, or remove the smart-button, or avoid the create button there.

The button at the component line level sends by context the product of that line.
The `mrp.document `model is not intended to create attachments that are not associated with a product (component).
In my opinion, the top smart-button is useful to see all attachments together from all components (useful if you have for example 20 components with attachments).

**Comment by pedrobaeza** [comment]
And why you can't send the same product in the context on the top smart-button?

**Comment by victoralmau** [comment]
Because the context that is currently sent is that of each of the components. The product from the list of materials could be sent by context in the upper smart-button, but I think it would be a bit strange.

**Comment by pedrobaeza** [comment]
OK, I see. Then can you avoid the attachment creation from that view? There's a way to do it, but I don't know which one right now.

**Comment by victoralmau** [comment]
Currently, from the top smart-button, the "upload" button is displayed but if we enter from the button of each component the "upload" button is displayed and attachments can be created.

**Comment by pedrobaeza** [comment]
Then can you disable the upload button from the top smart-button?

**Comment by victoralmau** [comment]
Yes, it is already disabled.

ejemplo-bom

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**